src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt

```diff
@@ -124,6 +136,63 @@ private suspend fun writeOutputVCF(
 
 }
 
+/**
+ * Writes the merged VCF files. This is used when a bedfile is provided.
+ * The bedfile is used to create one VCF file per range in the bedfile.
+ */
+private suspend fun writeOutputVCF(
+    outputFile: String,
+    bedfile: String,
+    samples: List<String>,
+    variantContextChannel: Channel<Deferred<List<VariantContext>>>
+) {
+
+    val ranges = readBedfile(bedfile)
+
+    val header = VCFHeader(createGenericVCFHeaders(samples))
+
+    var writePositionRange: PositionRange? = null
+    var writer: VariantContextWriter? = null
+    for (deferred in variantContextChannel) {
+
+        val variantContexts = deferred.await()
+        variantContexts.forEach { variantContext ->
+
+            val currentPositionRange = PositionRange(variantContext.contig, variantContext.start, variantContext.end)
+
+            // If the current position range doesn't overlap with the current write position range
+            // then close the current writer and open a new one
```

Comment:
What happens if the VariantContext record overlaps multiple ranges?  Should it get split, with parts written to range-specific files for all files it overlaps?